tests/retworkx_backwards_compat/digraph/test_core_number.py

```diff
@@ -68,29 +68,29 @@ def setUp(self):
 
     def test_directed_empty(self):
         digraph = retworkx.PyDiGraph()
-        res = retworkx.core_number(digraph)
+        res = retworkx.digraph_core_number(digraph)
```

Comment:
Keep the generic function call, it is good to test that our dispatcher works

Reply:
Just curious what the rationale is for this. All the other tests (except for `all_pairs_all_simple_paths`) call the `graph_` or `digraph_` version in `mod.rs`. It seems if you call `core_number` directly here, the `mod.rs` versions don't get tested.

Reply:
The mod.rs version gets tested because of the dispatch in `__init__.py`. You might want to check this: https://github.com/Qiskit/rustworkx/blob/main/rustworkx/__init__.py#L1087-L1116

Reply:
Also, some tests are older than #241 . That's why they didn't use it at the time

Reply:
Thanks. I got it now. I'll update the tests.